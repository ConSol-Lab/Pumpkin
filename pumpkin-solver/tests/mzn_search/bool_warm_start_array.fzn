var bool: x:: output_var;
var bool: y:: output_var;
var 0..2: X_INTRODUCED_0_:: output_var:: is_defined_var;
var 0..1: X_INTRODUCED_2_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_3_ ::var_is_introduced :: is_defined_var;
constraint int_lin_eq([1,1,-1],[X_INTRODUCED_3_,X_INTRODUCED_2_,X_INTRODUCED_0_],0):: defines_var(X_INTRODUCED_0_):: ctx_pos;
constraint bool2int(y,X_INTRODUCED_2_):: defines_var(X_INTRODUCED_2_);
constraint bool2int(x,X_INTRODUCED_3_):: defines_var(X_INTRODUCED_3_);
solve :: warm_start_array([warm_start_bool([x],[true]),warm_start_bool([y],[false])]) maximize X_INTRODUCED_0_;

