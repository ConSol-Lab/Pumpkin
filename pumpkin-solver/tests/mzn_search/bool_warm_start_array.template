var bool: x;
var bool: y;

array[1..2] of var bool: variables = [x, y];

solve :: warm_start_array([warm_start([x], [true]), warm_start([y], [false])]) maximize x + y;

