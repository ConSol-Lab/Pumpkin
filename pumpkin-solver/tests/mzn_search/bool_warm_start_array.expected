x = true;
y = false;
X_INTRODUCED_0_ = 1;
----------
x = true;
y = true;
X_INTRODUCED_0_ = 2;
----------
==========
